en/10_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -172,3 +172,7 @@ PHPDocs are not only useful when looking at the source code, but are also used i
 - Prefer the identical `===` operator over the equality `==` operator for comparisons.
 - If you directly reference a third-party dependency in code, then ensure the dependency is required in the module's `composer.json` file.
 - Avoid hardcoding values when there is a method available to dynamically get a value, for example use [`DataObjectSchema::tableName()`](api:SilverStripe\ORM\DataObjectSchema::tableName()) to get the table name for a `DataObject` model rather than hard coding it.
+- Do not use the `self` keyword, e.g. `self::myMethod()` instead use the short-name of the current class e.g. `MyClass::myMethod()` which is functionally equivalent. This avoids a specific class of bug that is introduced when using late static binding downstream from `self`.
+  - The one exception to this rule is traits as there is no way to know the class the trait is applied to.
+  - The use of `static` keyword is perfectly OK and is often appropriate.
+  - You should probably not use `MyClass::config()` and instead use `static::config()` as that will read any overriden config from subclasses. `self::config()` somewhat unintuitively behaves like `static::config()` and should not be used.
```

Comment:
I've validated that `self::config()` does behave like `static::config()` and not like `MyClass::config()` as would be intuitive.

**MyDataObject.php**
```php
<?php

use SilverStripe\Core\Config\Configurable;
use SilverStripe\ORM\DataObject;

class MyDataObject extends DataObject
{
    use Configurable;

    private static $table_name = 'MyDataObject';

    private static $something = 'one';

    public function test()
    {
        var_dump([
            self::config()->get('something'),
            static::config()->get('something'),
            self::config()->something,
            MyDataObject::config()->something,
        ]);
    }
}
```

**MyDataObjectChild.php**
```php
<?php

use SilverStripe\Core\Config\Configurable;

class MyDataObjectChild extends MyDataObject
{
    private static $table_name = 'MyDataObjectChild';

    private static $something = 'two';
}
```

```php
MyDataObjectChild::create()->test();
/*
  0 => string 'two'
  1 => string 'two'
  2 => string 'two'
  3 => string 'one'
*/
```